extension/ensureSaved.js

```diff
@@ -1,66 +1,51 @@
 let edited = false;
 
-const onEdit = () => {
-	edited = true
-};
+const onEdit = () => edited = true;
 
-const undoEdits = () => {
-	edited = false;
-};
+const undoEdits = () => edited = false;
 
-const addListener = (element) => {
-	element.addEventListener("change", onEdit);
-	element.addEventListener("keydown", onEdit);
-};
-
-const removeListener = (element) => {
-	element.removeListener("change", onEdit);
-	element.removeListener("keydown", onEdit);
-};
-
-const editListenersFrom = (parent, update) => () => {
-	const editableElements = getElementsByTags(parent, RELEVANT_TAGS);
-	editableElements.forEach(update);
-};
+const addEditListener = (parent) =>
+	getElementsByTags(parent, RELEVANT_TAGS).forEach((element) => {
+		element.addEventListener("change", onEdit);
+		element.addEventListener("keydown", onEdit);
+	});
 
 const addOnEditClicked = (element) =>
 	element.addEventListener("click", undoEdits);
 
 const observer = new MutationObserver((mutations) => {
-	addOnCancelEditButton();
+	/**
+	 * This potentially means things are added same listener to the same node multiple times,
+	 * but who cares if we set the edited boolean as true 8 times lmao
+	 */
+	addUndoEditListener();
 	mutations.forEach((mutation) =>
-		mutation.addedNodes.forEach((node) => {
-			editListenersFrom(node, addListener)();
-		}));
+		mutation.addedNodes.forEach(addEditListener));
 });
 
-const addOnCancelEditButton = () => [
+const addUndoEditListener = () => [
 	document.getElementById("book_editTabTextEditCancel1"),
 	document.getElementById("book_editTabTextEditCancel2"),
 	document.getElementById("book_editTabTextSave1"),
 	document.getElementById("book_editTabTextSave2"),
-	document.getElementById("book_editTabTextDelete"), // there's only one of these, so that it doesn't alert you when you're deleting something (?)
+	document.getElementById("book_editTabTextDelete"), // so that it doesn't alert you when you're deleting something (?)
 ].forEach(addOnEditClicked);
```

Comment:
this is a tad confuzzling -- we're trying to make it so that when these elements are clicked, we set `isEdited = false`, but `addOnEditClicked` doesn't really reflect that. might be a contender to rename in another PR (?)

Reply:
OH oops yeah I forgot to update that name one sec